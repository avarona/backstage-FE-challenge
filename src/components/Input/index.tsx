import React from 'react';
import styles from './styles.module.scss';

type Props = {
  value: string;
  onChange?: (e: React.FormEvent<HTMLInputElement>) => void;
}

const Input = ({ value, onChange }: Props) => (
  <div className={styles.inputContainer}>
    <input className={styles.input} type="text" name="input" onChange={onChange} value={value} />
  </div>
);

export default Input;
