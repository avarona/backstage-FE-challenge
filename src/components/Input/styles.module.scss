.inputContainer {
  width: 400px;
}

.input {
  width: 100%;
  padding: 10px;
  border-radius: 4px;
  border: 1px solid #aaa;
  outline: none;
}
