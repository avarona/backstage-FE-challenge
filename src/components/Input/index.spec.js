import React from 'react';
import { shallow } from 'enzyme';
import Input from './';

describe('Input', () => {
  const wrapper = shallow(<Input value="I'm an input" onChange={jest.fn()} />);

  it('renders', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('is an input', () => {
    expect(wrapper.find('input').exists()).toBe(true);
  });
});
