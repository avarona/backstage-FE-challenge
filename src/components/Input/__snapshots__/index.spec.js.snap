// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders 1`] = `"<div class=\\"inputContainer\\"><input type=\\"text\\" class=\\"input\\" pattern=\\"^[1-9][0-9]?$|^100$\\" name=\\"input\\" value=\\"I&#x27;m an input\\" placeholder=\\"Enter a number between 0 and 101\\" required=\\"\\"/></div>"`;
